set.mm

```diff
@@ -258381,6 +258381,34 @@ This function maps an element of the formal polynomial algebra (with
       KUMGVTWCWHWAKWBVLVMWSGVTWHBHUCVNVOVPVHVQ $.
   $}
 
+  ${
+    $d B x $.  $d N x $.  $d Q x $.  $d ph x $.
+    evlsgsumadd.q $e |- Q = ( ( I evalSub S ) ` R ) $.
+    evlsgsumadd.w $e |- W = ( I mPoly U ) $.
+    evlsgsumadd.0 $e |- .0. = ( 0g ` W ) $.
+    evlsgsumadd.u $e |- U = ( S |`s R ) $.
+    evlsgsumadd.p $e |- P = ( S ^s ( K ^m I ) ) $.
+    evlsgsumadd.k $e |- K = ( Base ` S ) $.
+    evlsgsumadd.b $e |- B = ( Base ` W ) $.
+    evlsgsumadd.i $e |- ( ph -> I e. V ) $.
+    evlsgsumadd.s $e |- ( ph -> S e. CRing ) $.
+    evlsgsumadd.r $e |- ( ph -> R e. ( SubRing ` S ) ) $.
+    evlsgsumadd.y $e |- ( ( ph /\ x e. N ) -> Y e. B ) $.
+    evlsgsumadd.n $e |- ( ph -> N C_ NN0 ) $.
+    evlsgsumadd.f $e |- ( ph -> ( x e. N |-> Y ) finSupp .0. ) $.
+    $( Polynomial evaluation maps (additive) group sums to group sums.
```

Comment:
It seems hypotheses of theorems in this section are sort of unordered.  Without knowing much about general conventions in algebra, I would prefer have them ordered such that one introducing a term is mentioned before (or consequently the other way round) another using that term.  So the primitive ones (like evlsgsumadd.i, evlsgsumadd.s, evlsgsumadd.n) come first, followed by evlsgsumadd.r, evlsgsumadd.k, and so on.

Reply:
I just went with what evls1gsumadd did, with the exception of moving `K` right after `P`

Based on looking at various theorems around, it seems like:

+ "variable definitions" go at the start
+ deductive hypotheses go at the end
+ structure definitions go before their components

I do agree that there is still some ambiguity with the ordering, even if this was an explicit convention somewhere

Reply:
I think the current ordering is fine:
* "variable definitions" are at the start
* followed by deductive hypotheses with membership
* followed by other deductive hypotheses